### PR TITLE
Details screen

### LINKED ISSUE #2: incorrect class name.
incorrct class name.

use  EmployeeListViewController

_Originally posted by @damencki in https://github.com/kenzywa/table-application/pull/1#discussion_r602860527_

### PR DESCRIPTION
**Pull request by kenzywa**

### CODE REVIEW

File: TableApplication.xcodeproj/xcuserdata/roman.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (damencki) [inline TableApplication.xcodeproj/xcuserdata/roman.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:1]: Add file to gitignore

File: TableApplication/Employee.swift [hunk 2]
```diff
@@ -3,5 +3,9 @@ import Foundation
 struct Employee {
     var name: String
     var surname: String
-    var age: Int
+    var age: String
```
Comment (damencki) [inline TableApplication/Employee.swift]: Incorrect type

File: TableApplication/Employee.swift [hunk 3]
```diff
@@ -3,5 +3,9 @@ import Foundation
...
+    var profession: String
+    var male: String
+    var numberPhone: String
```
Comment (damencki) [inline TableApplication/Employee.swift]: Inorrect type

File: TableApplication/SecondViewController.swift [hunk 4]
```diff
@@ -0,0 +1,94 @@
+import UIKit
+
+class SecondViewController: UIViewController {
```
Comment (damencki) [inline TableApplication/SecondViewController.swift]: Incorrect class name

File: TableApplication/SecondViewController.swift [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        self.view.addSubview(numberPhoneTextField)
+        
+        buttonToSave.center = CGPoint(x: 310, y: 235)
+        buttonToSave.setTitle("Save", for: .normal)
+        buttonToSave.layer.borderWidth = 2
+        buttonToSave.layer.cornerRadius = 8
+        buttonToSave.backgroundColor = .systemGreen
+        buttonToSave.setTitleColor(.black, for: .normal)
+        buttonToSave.addTarget(self, action: #selector(didTapSaveButton), for: .touchUpInside)
+        self.view.addSubview(buttonToSave)
```
Comment (damencki) [inline TableApplication/SecondViewController.swift]: extract to method

File: TableApplication/SecondViewController.swift [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    }
+    
+    @objc private func didTapSaveButton() {
+            guard let name = nameTextField.text,
+                  let surname = surnameTextField.text,
+                  let numberPhone = numberPhoneTextField.text,
+                  let id = employee?.id,
+                  let profession = professionTextField.text else {
+                return
```
Comment (damencki) [inline TableApplication/SecondViewController.swift]: Where is error handling?

File: TableApplication/ViewController.swift [hunk 7]
```diff
@@ -1,51 +1,64 @@
...
+                     Employee(name: "Marina", surname: "Bondarenko", age: "28", profession: "Sales Manager", male: "W", numberPhone: "+79811234567", id: UUID().uuidString),
+                     Employee(name: "Kirill", surname: "Unshikov", age: "21" , profession: "Go dev", male: "M", numberPhone: "+79811234567", id: UUID().uuidString),
+                     Employee(name: "Roman", surname: "Suhov", age: "22" , profession: "Python dev ", male: "m", numberPhone: "+79811234567", id: UUID().uuidString),
+                     Employee(name: "Raif", surname: "Garipov", age: "19" , profession: "Student ", male: "M", numberPhone: "+78005553535", id: UUID().uuidString),
+                     Employee(name: "Igor", surname: "Avgustov", age: "22" , profession: "Frontend dev", male: "M", numberPhone: "+78005553535", id: UUID().uuidString)]
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
+        self.title = "Table of Employees"
```
Comment (damencki) [inline TableApplication/ViewController.swift]: Create private struct with constants

File: TableApplication/ViewController.swift [hunk 8]
```diff
@@ -1,51 +1,64 @@
...
         self.firstTableView.delegate = self
         self.firstTableView.dataSource = self
-        
         firstTableView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
-        
         view.addSubview(firstTableView)
     }
-    
+    func toSaveData(name : String, surname : String, profession : String, numberPhone : String, id : String) {
+
```
Comment (damencki) [inline TableApplication/ViewController.swift]: delete empty line

File: TableApplication/ViewController.swift [hunk 9]
```diff
@@ -1,51 +1,64 @@
...
+            if let index = employees.firstIndex(where: {$0.id == id}) {
+                employees[index].name = name
+                employees[index].surname = surname
+                employees[index].numberPhone = numberPhone
+                employees[index].profession = profession
+                firstTableView.reloadData()
+            } else {
+                return
```
Comment (damencki) [inline TableApplication/ViewController.swift]: error handling?

File: TableApplication/ViewController.swift [hunk 10]
```diff
@@ -1,51 +1,64 @@
...
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let employee = employees[indexPath.row]
+        let secondVC = SecondViewController(delegate: self)
+        secondVC.employee = employee
+        navigationController?.pushViewController(secondVC, animated: true)
+    }
     //MARK: - UITableViewDelegate
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        return 60.0
+        return 80.0
```
Comment (damencki) [inline TableApplication/ViewController.swift]: struct with constants

File: TableApplication/delegateProtocol.swift [hunk 11]
```diff
@@ -0,0 +1,7 @@
+protocol DataDelegate {
```
Comment (damencki) [inline TableApplication/delegateProtocol.swift]: incorrect protocol name

**Review by damencki — CHANGES_REQUESTED** [review]

File: TableApplication/Employee.swift (hunk 2 above)
Comment (kenzywa) [reply TableApplication/Employee.swift]: Solved

File: TableApplication/Employee.swift (hunk 3 above)
Comment (kenzywa) [reply TableApplication/Employee.swift]: Solved

File: TableApplication/SecondViewController.swift (hunk 4 above)
Comment (kenzywa) [reply TableApplication/SecondViewController.swift]: Solved

File: TableApplication/ViewController.swift (hunk 10 above)
Comment (kenzywa) [reply TableApplication/ViewController.swift]: Solved

File: TableApplication/ViewController.swift (hunk 8 above)
Comment (kenzywa) [reply TableApplication/ViewController.swift]: Solved

File: TableApplication/ViewController.swift (hunk 7 above)
Comment (kenzywa) [reply TableApplication/ViewController.swift]: Solved

File: TableApplication/delegateProtocol.swift (hunk 11 above)
Comment (kenzywa) [reply TableApplication/delegateProtocol.swift]: Solved

File: TableApplication.xcodeproj/xcuserdata/roman.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist (hunk 1 above)
Comment (kenzywa) [reply TableApplication.xcodeproj/xcuserdata/roman.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:1]: Solved

File: TableApplication/SecondViewController.swift (hunk 5 above)
Comment (kenzywa) [reply TableApplication/SecondViewController.swift]: Solved

File: TableApplication/ViewController.swift (hunk 9 above)
Comment (kenzywa) [reply TableApplication/ViewController.swift]: Solved

File: TableApplication/TableViewController.swift [hunk 12]
```diff
@@ -0,0 +1,78 @@
+import UIKit
+
+class TableViewController: UIViewController, UITableViewDelegate, UITableViewDataSource, EmployeeEditorDelegate {
```
Comment (damencki) [inline TableApplication/TableViewController.swift]: incorrct class name.

use  EmployeeListViewController

**Review by damencki — COMMENTED** [review]
nice job

**Pull request merged**